src/tools/miri/src/helpers.rs

```diff
@@ -920,25 +920,12 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
 
     /// Check that the ABI is what we expect.
     fn check_abi<'a>(&self, fn_abi: &FnAbi<'tcx, Ty<'tcx>>, exp_abi: Conv) -> InterpResult<'a, ()> {
-        let caller_abi_name = match fn_abi.conv {
-            Conv::C => "C",
-            Conv::Rust => "Rust",
-            Conv::X86Stdcall => "X86Stdcall",
-            _ => { panic!("Unsupported caller ABI");}
-        }; 
-        let callee_abi_name = match exp_abi {
-            Conv::C => "C",
-            Conv::Rust => "Rust",
-            Conv::X86Stdcall => "X86Stdcall",
-            _ => { panic!("Unsupported callee ABI");}
-        }; 
-
         if fn_abi.conv != exp_abi {
             throw_ub_format!(
                 "calling a function with ABI {} using caller ABI {}",
-                callee_abi_name,
-                caller_abi_name
-            )
+                format!("{:?}", exp_abi),
+                format!("{:?}", fn_abi.conv)
+            );
         }
```

Comment:
Nested `format!` calls are entirely unnecessary. Just use `{:?}` in the main format string.